test/unit/storage/entities/block.js

```diff
@@ -0,0 +1,669 @@
+
+		describe('filters', () => {
+			// To make add/remove filters we add their tests.
+			it('should have only specific filters');
+			// For each filter type
+			it('should return matching result for provided filter');
+		});
+	});
+
+	describe('getOne()', () => {
+		let block;
+
+		before(done => {
+			const adapter = { loadSQLFile: sinonSandbox.stub() };
+			block = new Block(adapter);
+			done();
+		});
+
+		it('should accept valid filters', done => {
+			const filters = [{ height: 101 }, { timestamp_gte: 1234567890 }];
+			expect(() => {
+				block.getOne(filters);
+			}).to.not.throw(NonSupportedFilterTypeError);
+			done();
+		});
+
+		it('should throw error for invalid filters', done => {
+			const filters = [{ invalid_filter: 1 }, { timestamp_gte: 1234567890 }];
+			expect(() => {
+				block.getOne(filters);
+			}).to.throw(NonSupportedFilterTypeError);
+			done();
+		});
+
+		it('should accept valid options', done => {
+			const options = { limit: 100, offset: 0 };
+			expect(() => {
+				block.getOne({}, options);
+			}).to.not.throw(NonSupportedOptionError);
+			done();
+		});
+
+		it('should throw error for invalid options', done => {
+			const options = { invalid_option: 1, offset: 0 };
+			expect(() => {
+				block.getOne({}, options);
+			}).to.throw(NonSupportedOptionError);
+			done();
+		});
+
+		it(
+			'should call adapter.executeFile with proper param for FIELD_SET_SIMPLE'
+		);
+		it('should accept "tx" as last parameter and pass to adapter.executeFile');
+		it(
+			'should resolve with one object matching specification of type definition for FIELD_SET_SIMPLE'
+		);
+		it(
+			'should reject with error if matched with multiple records for provided filters'
+		);
+		it('should not change any of the provided parameter');
+
+		describe('filters', () => {
+			// To make add/remove filters we add their tests.
+			it('should have only specific filters');
+			// For each filter type
+			it('should return matching result for provided filter');
+		});
+	});
+
+	describe('create()', () => {
+		it('should accept only valid options');
+		it('should throw error for in-valid options');
+		it('should call getValuesSet with proper params');
+		it('should call adapter.executeFile with proper params');
+		it('should create a block object successfully');
+		it('should skip if any invalid attribute is provided');
+		it('should reject with invalid data provided');
+		it('should populate block object with default values');
+	});
+
+	describe('update()', () => {
+		it('should always throw NonSupportedOperationError', done => {
+			expect(Block.prototype.update).to.throw(NonSupportedOperationError);
+			done();
+		});
+	});
+
+	describe('updateOne()', () => {
+		it('should always throw NonSupportedOperationError', done => {
+			expect(Block.prototype.updateOne).to.throw(NonSupportedOperationError);
+			done();
+		});
+	});
+
+	describe('isPersisted()', () => {
+		let block;
+
+		before(done => {
+			const adapter = {
```

Comment:
What if we have only one adapter stub with all it's stubbed properties and gets reused for all tests and restored on afterEach() ?

Reply:
Some tests use different `adapter.executeFile` stub because it requires different `resolves()` scenarios